### PR TITLE
Automated fixed for AgentPool ControlIds

### PR DESCRIPTION
**Pull request by v-dpardeshi**
## Description
Automated Fixed for below Controlids :-
1. ADO_AgentPool_DP_No_Secrets_In_Capabilities
2. ADO_AgentPool_DP_Enable_Auto_Update

### CODE REVIEW

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 [hunk 1]
```diff
@@ -445,6 +508,40 @@ class AgentPool: ADOSVTBase
...
+                    $body =$CurrentAgent.CapabilitiesWithSecret | ConvertTo-Json
+    
+                }  
+                $url = "https://dev.azure.com/{0}/_apis/distributedtask/pools/{1}/agents/{2}/usercapabilities?api-version=5.0-preview.1" -f $($this.OrganizationContext.OrganizationName),$($CurrentAgent.PoolId), $($CurrentAgent.AgentId);          
+                $header = [WebRequestHelper]::GetAuthHeaderFromUriPatch($url)
+                $webRequestResult = Invoke-RestMethod -Uri $url -Method Put -ContentType "application/json" -Headers $header -Body $body	
+
+            }
+          						    
+            $controlResult.AddMessage([VerificationResult]::Fixed,  "User-defined capabilities for agent pool has been fixed.");
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1:536]: We should add a message that what all capabilities have been removed/added back

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 [hunk 2]
```diff
@@ -425,6 +479,15 @@ class AgentPool: ADOSVTBase
                         $display=($agentsWithSecretsInEnv | FT AgentName,Capabilities -AutoSize | Out-String -Width 512)
                         $controlResult.AddMessage($display)
                         $controlResult.SetStateData("Agent-wise list of user-defined capabilities with secrets: ", $agentsWithSecretsInEnv );
+                        $backupDataObject= @()
+                        @($agentDetails.Keys) | ForEach-Object {
+                            $key = $_
+                            $obj = '' | Select @{l="PoolId";e={$poolId}}, @{l="AgentId";e={$key}},@{l="CapabilitiesWithSecret";e={($agentDetails.item($key))[0]}}, @{l="CapabilitiesWithoutSecret";e={($agentDetails.item($key))[1]}}
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1:485]: We should not store the secret value in plain text on the local machine. Please encrypt it here, and decrypt it while using it in auto fix method. Also there seems to be some flaw in backupobj. Capabilities without secrets are also being added in withsecrets var:
"CapabilitiesWithoutSecret" : {
                    "test2" : "abc"
              }, 
              "CapabilitiesWithSecret" : {
                  "secret" : "Password123!@#", 
                  "test2" : "abc"
              }

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 [hunk 3]
```diff
@@ -212,6 +212,10 @@ class AgentPool: ADOSVTBase
                 else
                 {
                     $controlResult.AddMessage([VerificationResult]::Failed,"Auto-update of agents is disabled for [$($this.AgentPoolOrgObj.name)] agent pool.");
+                    if ($this.ControlFixBackupRequired) {
+                        #Data object that will be required to fix the control
+                        $controlResult.BackupControlState = $this.AgentPoolOrgObj;
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1:217]: whole $this.AgentPoolOrgObj object is not needed to store

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 [hunk 4]
```diff
@@ -229,6 +233,49 @@ class AgentPool: ADOSVTBase
...
+        {
+            #Backup data object is not required in this scenario.
+            $RawDataObjForControlFix = @();
+            $RawDataObjForControlFix = ([ControlHelper]::ControlFixBackup | where-object {$_.ResourceId -eq $this.ResourceId}).DataObject
+
+            $body = ""
+
+            if (-not $this.UndoFix)
+            {                 
+                if ($body.length -gt 1) {$body += ","}
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1:248]: ?? if ($body.length -gt 1) this condition will never come true here, just above 2 line $body is declared as $body = ""

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 [hunk 5]
```diff
@@ -445,6 +508,40 @@ class AgentPool: ADOSVTBase
...
+                if (-not $this.UndoFix)
+                {                 
+                    $body =  $CurrentAgent.CapabilitiesWithOutSecret | ConvertTo-Json
+                }
+                else 
+                {
+                    $body =$CurrentAgent.CapabilitiesWithSecret | ConvertTo-Json
+    
+                }  
+                $url = "https://dev.azure.com/{0}/_apis/distributedtask/pools/{1}/agents/{2}/usercapabilities?api-version=5.0-preview.1" -f $($this.OrganizationContext.OrganizationName),$($CurrentAgent.PoolId), $($CurrentAgent.AgentId);
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1:530]: $this.OrganizationContext.OrganizationName , $CurrentAgent.PoolId, $CurrentAgent.AgentId; is enough, $($ not needed

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 [hunk 6]
```diff
@@ -229,6 +233,47 @@ class AgentPool: ADOSVTBase
...
+                    "id":$($RawDataObjForControlFix),
+                    "autoUpdate":false
+                }
+"@;
+
+            }  
+            $url = " https://dev.azure.com/{0}/_apis/distributedtask/pools/{1}?api-version=5.0-preview.1" -f $($this.OrganizationContext.OrganizationName),$($RawDataObjForControlFix);          
+			$header = [WebRequestHelper]::GetAuthHeaderFromUriPatch($url)
+            $webRequestResult = Invoke-RestMethod -Uri $url -Method Patch -ContentType "application/json" -Headers $header -Body $body							    
+            $controlResult.AddMessage([VerificationResult]::Fixed,  "Auto-Update setting for agent pool have been changed.");
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1:268]: has been changed

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 [hunk 7]
```diff
@@ -445,6 +509,50 @@ class AgentPool: ADOSVTBase
...
+                $CurrentAgent= $_
+                if (-not $this.UndoFix)
+                {                 
+                    $body =  $FixObjTable | ConvertTo-Json   
+                    $controlResult.AddMessage([VerificationResult]::Fixed,  "Following User-defined capabilities for agent pool has been Removed.");      
+                }
+                else 
+                {
+                    $body = $UndoFixObjTable  | ConvertTo-Json
+                    $controlResult.AddMessage([VerificationResult]::Fixed,  "Following User-defined capabilities for agent pool has been Added.");
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1:541]: user-defined..... have been added: (uniform capitalization and colon in the end)

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 [hunk 8]
```diff
@@ -445,6 +509,50 @@ class AgentPool: ADOSVTBase
...
+            $display = ($UndoFixObjTable.Keys |  FT -AutoSize | Out-String -Width 512);
+            #The api requires both secret and non secret capabilities in order to update on the agent
+            $UndoFixObjTable += $FixObjTable 
+
+            $RawDataObjForControlFix | ForEach-Object {
+                $CurrentAgent= $_
+                if (-not $this.UndoFix)
+                {                 
+                    $body =  $FixObjTable | ConvertTo-Json   
+                    $controlResult.AddMessage([VerificationResult]::Fixed,  "Following User-defined capabilities for agent pool has been Removed.");
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1:536]: user-defined..... have been removed:

**Review by juhi037 — APPROVED** [review]

**Review by arvindsingh-cloud — APPROVED** [review]

**Review by v-soukum — APPROVED** [review]

**Review by SaikumarRagala — APPROVED** [review]

**Pull request merged**